build.gradle

```diff
@@ -88,94 +76,11 @@ if (hasProperty('buildScan')) {
     }
 }
 
-// Linter
-spotless {
-    java {
-        googleJavaFormat()
-        targetExclude "${buildDir}/**"
-    }
-}
-
-compileJava.dependsOn spotlessApply
-
-// Static analysis
-spotbugs {
-    effort = 'max'
-    // This makes the "html" reports come out in plain text so you can just open the file in IntelliJ
-    // and look at your bugs instead of having to switch to a browser.
-    extraArgs = ['-emacs']
-}
-spotbugsMain {
-    reports {
-        html {
-            enabled = true
-            destination = file("$buildDir/reports/spotbugs/main.html")
-        }
-    }
-}
-spotbugsTest {
-    reports {
-        html {
-            enabled = true
-            destination = file("$buildDir/reports/spotbugs/test.html")
-        }
-    }
-}
-
-tasks.withType(Test) {
-    testLogging {
-        events = ["passed", "failed", "skipped"]
-        // Causes the correct line to be reported on an exception.
-        exceptionFormat = org.gradle.api.tasks.testing.logging.TestExceptionFormat.FULL
-        reports {
-            // Write XML file (used by CircleCI, Jenkins, etc) to build/test-results/*
-            junitXml.enabled = true
-            // Write human-readable test report to build/reports/tests/*
-            html.enabled = true
-        }
-    }
-}
-
-test {
-    useJUnitPlatform()
-}
-
-task unitTest(type: Test) {
-    useJUnitPlatform {
-        includeTags 'unit'
-    }
-}
-
-// TODO(gjordan): turn off doclint to remove "no @return value" warnings
-// https://blog.joda.org/2014/02/turning-off-doclint-in-jdk-8-javadoc.html
-java {
-    withJavadocJar()
-}
-
-def artifactory_repo_key = System.getenv("ARTIFACTORY_REPO_KEY") != null ? System.getenv("ARTIFACTORY_REPO_KEY") : 'libs-snapshot-local'
-def artifactory_username = System.getenv('ARTIFACTORY_USERNAME')
-def artifactory_password = System.getenv('ARTIFACTORY_PASSWORD')
+apply from: "$rootDir/gradle/dependency-locking.gradle"
+apply from: "$rootDir/gradle/javadoc.gradle"
+apply from: "$rootDir/gradle/publishing.gradle"
+apply from: "$rootDir/gradle/quality.gradle"
```

Comment:
Is the idea that I'll be able to do an external apply from in the other repo and pick up this file? I'd still have duplicate xml configs.

Reply:
The idea here is to make our main Gradle file less monolithic, with clearer separation of concerns in each sub-config. It's purely about local organization. It will encourage cleaner copy-pasting of Gradle config between repos for as long as we're in that paradigm. Longer term, it seems clear that providing a central Gradle plugin is the right way to share config like this.